projectq/backends/_sim/_cppkernels/simulator.hpp

```diff
@@ -500,6 +568,12 @@ class Simulator{
     unsigned fusion_qubits_min_, fusion_qubits_max_;
     RndEngine rnd_eng_;
     std::function<double()> rng_;
+
+    // large array buffers to avoid costly reallocations
+    static StateVector tmpBuff1_, tmpBuff2_;
```

Comment:
In our use case we run the simulator inside of an optimization loop, so we create a new python object for the simulator a lot of times.
So this reduces the overhead for allocating memory for multiple instantiations of the simulator...
As python cannot run multiple (native) threads due to the GIL, I thought it would be ok to have static variables here. We can also remove the "static" here and modify our code so it reuses the same instance of the simulator...

Reply:
It's true that with the properties of the GIL, those static variables should not pose any issues.